### DISCUSSION TITLE
How to run examples?  (" ../build/plugins/python/py_tutor.py failed with error 404")

### DISCUSSION THREAD

**Discussion by alewissocal** [pyscript/pyscript#1259] [Q&A]
I am just starting, trying to run the examples locally. I successfully have the local python http.server running. When I open the local url, the browser (MS Edge) displays the example files I copied into a directory (hello_world.html, panel_kmeans.html, pandas.html). Clicking on any of them will partially execute, but then fail with the 404 error above. (See screenshot below)

Googling for "py_tutor.py" does not turn up anything except a few links to this github, so I am stuck. What am I doing wrong?

-alan 

![error](https://user-images.githubusercontent.com/127152950/223277766-f4990bf3-51c1-42d6-b799-dd093c80ac56.jpg)

**Comment by marimeireles** [comment]
Hey @alewissocal, welcome :)
Well, I guess the path to the file `py_tutor.py` is incorrect.
One of the files you're trying to open is linking to it but it can't find it.
If you're running the examples I can recommend using `make examples`. If you do that a new `examples` folder will show up under `pyscriptjs/examples` and those are going to have the correct linking for the local version you've built of pyscript.

You might find these two documents specially enlightening:
https://github.com/pyscript/pyscript/blob/main/docs/development/developing.md
https://github.com/pyscript/pyscript/blob/main/CONTRIBUTING.md

If you want to have a sync conversation don't hesitate to join the discord channel. But feel free to keep posting here if more questions show up!

Thanks for opening the discussion, cheers!

**Reply by alewissocal** [reply]
Thanks, Mariana, for the very helpful guide to building the project locally. This is something I will likely want to do.
For the moment, I have a follow-up on my original question.

This morning, I tried running the hello_world.html example at
 https://www.w3schools.com/html/tryit.asp?filename=tryhtml_default_default
Got the same error.

This raises an issue for me on the foundational goals of the project.
I would ultimately like to build a sort of dashboard-type application in python, and then be able to deploy it to the web on a server via pyscript, say at AWS. Is this going to be possible? Will it require also 'making' the project at the destination server? If not, then why does the hello_world example fail at www.w3schools?

Thanks again for your help.
-alan

**Comment by markqiu** [comment]
1. install nodejs environment
2. cd pyscriptjs
3. make run_examples

hope it can help

**Reply by alewissocal** [reply]
Thanks. After some effort, I got the project built locally and can run the examples. I see that some examples can be run from the examples page that appears in the browser and some require inserting the example-file-name after localhost:8080/. Is there some reason for this?

As Mariana hinted, I now see that 'py_tutor.py' is created locally as a consequence of 'make examples'. I get the sense that this file is critical to the project. But, as a complete novice, I don't have the background to understand the code in it. Can someone give a high-level overview of what py_tutor.py does or point me to some documentation about it?

Thanks again for the help.

**Comment by chisarie** [comment]
Hi, I actually see the same error appear when I open any of the demo examples on https://pyscript.net/examples/. Did I miss something? Or maybe is something wrong with the demo examples? I tried both on Firefox and Chrome

**Comment by marimeireles** [comment]
No, there's something wrong for sure.
Sorry @alewissocal I gave you the wrong answer. I thought it was something on your side but it's definitely a bug on ours.
I'm gonna investigate for a bit, see what I can find.
Thanks again for reporting people.
